lte/gateway/c/core/oai/test/amf/test_amf_encode_decode.cpp

```diff
@@ -880,6 +907,372 @@ TEST(test_dnn, test_amf_validate_dnn) {
   EXPECT_TRUE(rc == RETURNok);
 }
 
+class AmfUeContextTestServiceRequestProc : public ::testing::Test {
+ protected:
+#define MCC_DIGIT1 2
+#define MCC_DIGIT2 2
+#define MCC_DIGIT3 2
+#define MNC_DIGIT1 4
+#define MNC_DIGIT2 5
+#define MNC_DIGIT3 6
+#define IMSI64 222456000000101
+#define M_TMSI 0X212e5025
+#define AMF_SET_ID 1
+#define AMF_POINTER 0
+#define AMF_REGION_ID 1
+#define AMF_TAC 0x03
+
+  ue_m5gmm_context_s* ue_context;
+  amf_app_desc_t* amf_app_desc_p;
+  guti_m5_t guti;
+  tai_t tai;
+  const amf_ue_ngap_id_t AMF_UE_NGAP_ID = 0x05;
+  const gnb_ue_ngap_id_t gNB_UE_NGAP_ID = 0x09;
+  const uint32_t gnb_id                 = 0x01;
+  const imsi64_t imis64                 = IMSI64;
+  const uint32_t m_imsi                 = M_TMSI;
+
+  virtual void SetUp() {
+    itti_init(
+        TASK_MAX, THREAD_MAX, MESSAGES_ID_MAX, tasks_info, messages_info, NULL,
+        NULL);
+    amf_config_init(&amf_config);
+    amf_nas_state_init(&amf_config);
+
+    ue_context     = amf_create_new_ue_context();
+    amf_app_desc_p = get_amf_nas_state(false);
+
+    // insert ue context
+    if (ue_context) {
+      amf_insert_ue_context(AMF_UE_NGAP_ID, ue_context);
+    }
+
+    // imsi64
+    ue_context->amf_context.imsi64 = imis64;
+    // ue security context
+    ue_context->amf_context.member_present_mask |= AMF_CTXT_MEMBER_SECURITY;
+    // ueContextReq
+    ue_context->ue_context_request = M5G_UEContextRequest_requested;
+    // ue state
+    ue_context->mm_state = REGISTERED_IDLE;
+    // 5G TMSI
+    ue_context->amf_context.m5_guti.m_tmsi = m_imsi;
```

Comment:
move to a utility function

Reply:
default config already part of setup method which is to be used for setting configs , initialization etc..